continuationsPlugin/src/main/scala/continuations/CallsContinuationResumeWith.scala

```diff
@@ -26,8 +26,15 @@ private[continuations] object CallsContinuationResumeWith:
         .filterSubTrees {
           case Inlined(fun, _, _) =>
             fun.denot.matches(requiredMethod(suspendContinuationFullName))
+          case Return(_, _) => true
           case _ => false
         }
+        .takeWhile {
+          case Return(Inlined(fun, _, _), _) =>
+            fun.denot.matches(requiredMethod(suspendContinuationFullName))
+          case Return(_, _) => false
+          case _ => true
+        }
```

Comment:
Hmm. Wouldn't this fail on something like:

```scala
def mySuspend(x: String)(using Suspend): String =
  if(x == "hello") {
    return x
  } else {
    Continuation.suspend[String](c => c.resumeWith(Right("hello")))
  }
```
where the else is reachable, but there is an early return?

Reply:
Correct, I'm aware. It would also fail with any similar construct like:

```scala
def mySuspend(x: Int)(using Suspend): Int =
  while (x < 3) {
    return 47
  }
  Continuation.suspendContinuation[Int] { continuation =>
    continuation.resume(Right(1))
  }
```

I need to check if the `return` occurs at the outermost level of the function and filter only that case. I'm still figuring out this.